### PR TITLE
PWX-21277: Add callhome test

### PR DESCRIPTION
**Pull request by pault84**
Signed-off-by: Paul Theunis <paul@portworx.com>

**What this PR does / why we need it**:
Test to see that FA/FB essentials keeps working if we disable call-home

**Special notes for your reviewer**:
Waiting to get a positive test run.

http://jenkins.pwx.dev.purestorage.com/view/Control%20Plane/job/Torpedo/job/2-tp-nextpx-pure-esse/36

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -2380,6 +2380,57 @@ func (d *portworx) GetLicenseSummary() (torpedovolume.LicenseSummary, error) {
...
+		Sudo:            true,
+	}
+
+	var rtopts string
+	for k, v := range rtOpts {
+		rtopts += k + "=" + v + ","
+	}
+
+	rtopts = strings.TrimSuffix(rtopts, ",")
+	cmd := fmt.Sprintf("pxctl cluster options update --runtime-options %s", rtopts)
```
Comment (vinayakshnd) [inline drivers/volume/portworx/portworx.go:2399]: Do we have SDK API to do this?

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -2380,6 +2380,57 @@ func (d *portworx) GetLicenseSummary() (torpedovolume.LicenseSummary, error) {
...
+	if !enabled {
+		cmd = "pxctl sv call-home disable"
+	}
+
+	out, err := d.nodeDriver.RunCommand(n, cmd, opts)
+	if err != nil {
+		return fmt.Errorf("failed to toggle call-home, Err: %v %v", err, out)
+	}
+
+	logrus.Debugf("Successfully toggled call-home")
```
Comment (vinayakshnd) [inline drivers/volume/portworx/portworx.go:2430]: Should we print in debug whether we disabled or enabled call-home successfully?

File: drivers/volume/portworx/portworx.go [hunk 3]
```diff
@@ -2380,6 +2380,57 @@ func (d *portworx) GetLicenseSummary() (torpedovolume.LicenseSummary, error) {
...
+	opts := node.ConnectionOpts{
+		IgnoreError:     false,
+		TimeBeforeRetry: defaultRetryInterval,
+		Timeout:         defaultTimeout,
+		Sudo:            true,
+	}
+
+	cmd := "pxctl sv call-home enable"
+	if !enabled {
+		cmd = "pxctl sv call-home disable"
```
Comment (vinayakshnd) [inline drivers/volume/portworx/portworx.go:2422]: Is SDK API available for this?

File: tests/license/license_test.go [hunk 4]
```diff
@@ -577,6 +577,70 @@ var _ = Describe("{DeleteSecretRebootAllNodes}", func() {
 	})
 })
 
+// This test performs basic test of starting an application and destroying it (along with storage)
```
Comment (vinayakshnd) [inline tests/license/license_test.go:580]: Please change the comment to reflect the purpose of the test.

File: tests/license/license_test.go [hunk 5]
```diff
@@ -577,6 +577,70 @@ var _ = Describe("{DeleteSecretRebootAllNodes}", func() {
...
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("setupteardown-license-callhome-%d", i))...)
+		}
+		ValidateApplications(contexts)
+
+		opts := make(map[string]bool)
+		opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+
+		currNode := node.GetNodes()[0]
+		Step(fmt.Sprintf("Set License expiry timeout to 1 hour"), func() {
+			err := Inst().V.SetClusterOpts(currNode, map[string]string{
```
Comment (vinayakshnd) [inline tests/license/license_test.go:595]: Do we need to restart PX service after setting Cluster options?

File: drivers/volume/portworx/portworx.go (hunk 1 above)
Comment (pault84) [reply drivers/volume/portworx/portworx.go:2399]: Nope

File: drivers/volume/portworx/portworx.go (hunk 3 above)
Comment (pault84) [reply drivers/volume/portworx/portworx.go:2422]: nope

File: tests/license/license_test.go (hunk 5 above)
Comment (pault84) [reply tests/license/license_test.go:595]: i don't believe so?

Comment (pault84) [reply tests/license/license_test.go:595]: i'll add a reboot just in case

**Review by vinayakshnd — APPROVED** [review]
LGTM. provided minor comments.

File: tests/license/license_test.go [hunk 6]
```diff
@@ -577,6 +577,116 @@ var _ = Describe("{DeleteSecretRebootAllNodes}", func() {
 	})
 })
 
+// This test performs basic test disabling callhome and checking if the licnse stays valid
+var _ = Describe("{DisableCallHomeTest}", func() {
+	var contexts []*scheduler.Context
+	It("has to setup, validate and teardown apps", func() {
```
[the remark this answers is not in the document]
Comment (pault84) [reply tests/license/license_test.go:583]: Sure thing

**Pull request merged**